artifacts/artifact_funcs.jl

```diff
@@ -59,3 +60,28 @@ function pr_obs_data_path()
     )
     return AW.get_data_folder(pr_obs_data)
 end
+
+"""
+    artifact_data(datapath, datafile, date0=nothing, time_start=nothing, time_end=nothing)
+
+Truncates given data set, and constructs a new dataset containing only the dates that are used in the simulation
+"""
+function artifact_data(
+    datapath_full,
+    name,
+    datapath_trunc = nothing,
+    date0 = nothing,
+    time_start = nothing,
+    time_end = nothing,
+)
+    if datapath_trunc === nothing || date0 === nothing || time_start === nothing || time_end === nothing
```

Comment:
you can also dispatch off these types in the function signature using `::Nothing`, and have two separate methods instead of the if/else